src/random-seed/random-seed.c

```diff
@@ -16,10 +17,88 @@
 #include "mkdir.h"
 #include "string-util.h"
 #include "util.h"
+#include "terminal-util.h"
+
+static enum {
+        ACTION_NONE,
+        ACTION_LOAD,
+        ACTION_SAVE,
+} arg_action = ACTION_NONE;
 
 #define POOL_SIZE_MIN 512
 #define POOL_SIZE_MAX (10*1024*1024)
 
+static int help(void) {
+        _cleanup_free_ char *link = NULL;
+        int r;
+
+        r = terminal_urlify_man("systemd-random-seed", "8", &link);
+        if (r < 0)
+                return log_oom();
+
+        printf("systemd-random-seed [OPTIONS...] load\n"
+               "systemd-random-seed save\n\n"
+               "Load and save the system random seed at boot and shutdown\n\n"
+               "  -h --help                       Show this help\n"
+               "     --version                    Show package version\n"
+               "\nSee the %s for details.\n"
+               , link
+        );
+
+        return 0;
+}
+
+static int parse_argv(int argc, char *argv[]) {
+
+        enum {
+                ARG_VERSION = 0x100,
+        };
+
+        static const struct option options[] = {
+                { "help",               no_argument,       NULL, 'h'                    },
+                { "version",            no_argument,       NULL, ARG_VERSION            },
+                {},
+        };
+
+        int c;
+
+        assert(argc >= 0);
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "h", options, NULL)) >= 0)
+
+                switch (c) {
+
+                case 'h':
+                        return help();
+
+                case ARG_VERSION:
+                        return version();
+
+                case '?':
+                        return -EINVAL;
+
+                default:
+                        assert_not_reached("Unhandled option");
+                }
+
+        if (optind + 1 != argc) {
+                log_error("This program requires one argument.");
+                return -EINVAL;
+        }
+        if (streq(argv[optind], "load"))
+                arg_action = ACTION_LOAD;
+        else if (streq(argv[optind], "save"))
+                arg_action = ACTION_SAVE;
+        else {
+                log_error("Unknown verb '%s'.", argv[optind]);
+                return -EINVAL;
+        }
```

Comment:
(i appreciate the getoptification, but in an ideal world we'd go even further and use verbs.[ch] for dispatching the verbs here... but i guess this is unrelated to this PR)

Reply:
Yeah, and if the interface ends up being an env var, this goes away, so I won't spend too much time polishing this.